crates/core_arch/src/x86/avx512f.rs

```diff
@@ -54375,9 +54387,9 @@ mod tests {
 
     #[simd_test(enable = "avx512f")]
     unsafe fn test_mm512_stream_ps() {
-        #[repr(align(32))]
+        #[repr(align(64))]
         struct Memory {
-            pub data: [f32; 16],
+            pub data: [f32; 16], // 64 bytes
         }
```

Comment:
This test should have failed many times already. The only explanation I have for why that did not happen is that maybe LLVM optimizes away the entire test...

Reply:
Maybe the whole stack frame gets 64-byte aligned, since there are `__m512` values involved.

Reply:
If that happened it would also happen with this PR.

Reply:
The stack frame is probably different because you are missing the `nostack` option on the inline assembly. In fact these assembly blocks should include both `nostack` and `preserves_flags`.

Reply:
I'm pretty sure the test was just buggy before this PR. It doesn't actually ensure that the data is properly aligned.

Reply:
> In fact these assembly blocks should include both nostack and preserves_flags.

I have added the options in the last commit.